quelpa.el

```diff
@@ -882,7 +894,16 @@ Return a cons cell whose `car' is the root and whose `cdr' is the repository."
         (when (file-exists-p dir)
           (delete-directory dir t))
         (quelpa-build--princ-checkout repo dir)
-        (quelpa-build--run-process nil "git" "clone" repo dir)))
+        (cond
+         ((integerp quelpa-git-clone-depth)
+          (quelpa-build--run-process
+           nil "git" "clone"
+           "--depth" (int-to-string quelpa-git-clone-depth)
```

Comment:
If `quelpa-build--run-process` were enhanced to omit nil arguments, this `cond` could be omitted, and something like this could work:
```el
(quelpa-build--run-process
           nil "git" "clone"
           (when quelpa-git-clone-depth) 
             (list "--depth" (number-to-string quelpa-git-clone-depth)))
           repo dir)
```
This is how I call commands in `magit-todos`'s scanner definitions, and it makes for very understandable code.